database/scrapers/building_scrapper.ts

```
@@ -0,0 +1,158 @@
+
+      const formattedBuildingData = await Promise.all(
+        buildingsData.map(async (data) => {
+          const addressArray = data.addres.split(",");
+          return {
+            id: data.id,
+            identifier: data.name,
+            specialName: data.naturalName,
+            iconType: BuildingIcon.Icon,
+            campusId: -1,
+            addressLine1: addressArray.shift(),
+            addressLine2: addressArray.shift(),
+            latitude: data.latitude,
+            longitude: data.longitude,
+            haveFood: data.food ?? false,
+            cover: await this.uploadAndGetKey(data),
+            externalDigitalGuideMode: data.externalDigitalGuideMode,
+            externalDigitalGuideIdOrURL: data.externalDigitalGuideIdOrURL,
+            createdAt: resolveDate(data.createdAt),
+            updatedAt: resolveDate(data.updatedAt),
+          };
+        }),
+      );
+      for (const buildingEntry of formattedBuildingData) {
+        const campus = campusesMap.get(buildingEntry.identifier);
+        if (campus === undefined) {
+          this.logger.error(
+            `No campus assigned to building ${buildingEntry.identifier}`,
+          );
+        } else {
+          // update campuses
+          campus.cover = buildingEntry.cover;
+          buildingEntry.campusId = campus.id;
+          await campus.save();
+        }
+      }
+      await Building.createMany(formattedBuildingData);
+      this.logger.info("Buildings created !");
+      task.update(`finished running ${this.name} scrapper`);
+    } catch (err) {
+      this.logger.error("error within buildings and campuses scrapper");
+      throw new Error(`${err}`);
+    }
+  }
+
+  private async uploadAndGetKey(data: BuildingsData): Promise<string> {
+    try {
+      const imageKey = data.cover;
+      if (imageKey === null) {
+        this.logger.warning(`no image for building: [${data.id}]`);
+        return "";
+      }
```

don't return fake values, either throw an error, or make `null` a valid value.